src/mono/mono/mini/method-to-ir.c

```diff
@@ -11106,6 +11106,17 @@ mono_method_to_ir (MonoCompile *cfg, MonoMethod *method, MonoBasicBlock *start_b
 				}
 			}
 
+			/* UnmanagedCallersOnlyAttribute means ldftn should return a method callable from native */
+			if (G_UNLIKELY ((cmethod->flags & METHOD_ATTRIBUTE_PINVOKE_IMPL) == 0 &&
+					cmethod->wrapper_type == MONO_WRAPPER_NONE &&
+					mono_method_has_unmanaged_callers_only_attribute (cmethod))) {
+				MonoClass *delegate_klass = NULL;
+				MonoGCHandle target_handle = 0;
+				cmethod = mono_marshal_get_managed_wrapper (cmethod, delegate_klass, target_handle, cfg->error);
+				CHECK_CFG_ERROR;
```

Comment:
This isn't quite right.  On Mono we will throw when JITing the method that has the `ldftn` instruction that has a disallowed `cmethod` (e.g. an instance method).  On CoreCLR they throw when the method runs.  So we would need to capture the error and emit a throw here.

And probably something similar in the interpreter